packages/taquito/src/contract/contract-methods/contract-on-chain-view.ts

```diff
@@ -84,26 +46,15 @@ export class OnChainView {
    */
   async executeView(executionContext: ExecutionContextParams) {
     this.verifyContextExecution(executionContext);
-    const balance = (await this._readProvider.getBalance(this._contractAddress, 'head')).toString();
     const chainId = await this._readProvider.getChainId();
     const storage = await this._readProvider.getStorage(this._contractAddress, 'head');
-    return this.executeViewAndDecodeResult(
-      runCodeHelper(
-        this._smartContractViewSchema.viewArgsType,
-        this._smartContractViewSchema.viewReturnType,
-        this._contractStorageType,
-        this.adaptViewCodeToContext(
-          this._smartContractViewSchema.instructions,
-          executionContext.viewCaller,
-          balance
-        ),
-        this.transformArgsToMichelson(),
-        storage,
-        balance,
-        chainId,
-        executionContext.source
-      )
-    );
+    const viewArgs = this.transformArgsToMichelson();
+    return this.executeViewAndDecodeResult({
+      contract: this._contractAddress,
+      view: this._smartContractViewSchema.viewName,
+      input: { prim: 'Pair', args: [viewArgs, storage] },
```

Comment:
the integration test `contract-execute-on-chain-view.spec.ts` is failing.
I think that `input` should be `viewArgs` not `{ prim: 'Pair', args: [viewArgs, storage] }`

Also I think that `ExecutionContextParams`should be used in the params of `runScriptView`:
payer might be the ExecutionContextParams.viewCaller
and source the ExecutionContextParams.source